app/src/main/java/com/android/solvit/shared/model/request/ServiceRequestViewModel.kt

```diff
@@ -77,4 +145,43 @@ open class ServiceRequestViewModel(private val repository: ServiceRequestReposit
   fun selectRequest(serviceRequest: ServiceRequest) {
     _selectedRequest.value = serviceRequest
   }
+
+  fun unConfirmRequest(serviceRequest: ServiceRequest) {
+    saveServiceRequest(serviceRequest.copy(status = ServiceRequestStatus.PENDING))
+  }
+
+  fun confirmRequest(serviceRequest: ServiceRequest) {
+    saveServiceRequest(serviceRequest.copy(status = ServiceRequestStatus.ACCEPTED))
+  }
+
+  fun scheduleRequest(serviceRequest: ServiceRequest) {
+    saveServiceRequest(serviceRequest.copy(status = ServiceRequestStatus.SCHEDULED))
+  }
+
+  fun completeRequest(serviceRequest: ServiceRequest) {
+    saveServiceRequest(serviceRequest.copy(status = ServiceRequestStatus.COMPLETED))
+  }
+
+  fun archiveRequest(serviceRequest: ServiceRequest) {
+    saveServiceRequest(serviceRequest.copy(status = ServiceRequestStatus.ARCHIVED))
+  }
+
+  fun getTodayScheduledRequests(): List<ServiceRequest> {
+    val dateFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy", Locale.getDefault())
+    val timeFormatter = DateTimeFormatter.ofPattern("HH:mm", Locale.getDefault())
+    val today = LocalDate.now().format(dateFormatter)
+    return _scheduledRequests.value
```

Comment:
If _scheduledRequests.value contains a large number of elements, the repeated calls to .toDate(), .toInstant(), and .atZone() within both the filter and sortedBy clauses can degrade performance.